kuma/users/views.py

```diff
@@ -745,6 +764,36 @@ def recover(request, uidb64=None, token=None):
     return render(request, 'users/recover_failed.html')
 
 
+@require_POST
+def create_stripe_subscription(request):
+    user = request.user
+
+    error_message = None
+    try:
+        customer = None
+        if not user.stripe_customer_id:
+            customer = stripe.Customer.create(
+                email=request.POST.get('stripe_email', ''),
+                source=request.POST.get('stripe_token', ''),
+            )
+            user.stripe_customer_id = customer.id
+            user.save()
+        else:
+            customer = stripe.Customer.retrieve(user.stripe_customer_id)
+
+        if retrieve_stripe_subscription(customer) is None:
+            stripe.Subscription.create(
+                customer=customer.id,
+                items=[{
+                    'plan': settings.STRIPE_PLAN_ID,
+                }]
+            )
+    except stripe.error.StripeError as e:
+        error_message = e.message
+
+    return redirect(reverse('users.user_edit', args=[user.username]), error_message)
```

Comment:
I don't understand this line. 
First of all, I think the more succinct syntax is to use `return redirect('users.user_edit', request.user.username)` as demonstrated on line 386. 
Second, why pass the `error_message` as a second parameter to the `redirect` function? What does that do? It's either a `None` or a string(?) but I don't know what it does. Is it an attempt to include the message to the new redirect URL? If so, shouldn't it be *something like* this:
```python
return redirect(reverse('users.user_edit', args=[user.username]) + f'?error_message={url_quote(error_message)')
```

Anyway, I think the whole approach should not be to just keep the `error_message` and pretend things are fine. 

In [my case](https://github.com/mdn/kuma/pull/6269#issuecomment-569982844) I think the ultimate error is because I have something wrong with my setup (`stripe.error.AuthenticationError: You did not provide an API key. You need to provide ...`). That shouldn't be captured. It should be a full exception that gets sent to Sentry. 

If there's some other error that's failing in the communication between our server and the Stripe server, that should be caught, logged, and ultimately displayed and explained to the user. 

Similar to how a form POST submission leads to a validation error, the page should return to the form and display the error in red. I mock-faked something:
<img width="764" alt="Screen Shot 2019-12-31 at 3 17 54 PM" src="https://user-images.githubusercontent.com/26739/71633196-c4a46c00-2be0-11ea-82df-69eadff5d399.png">

Reply:
And if that happens, I think we should use `raven` to manually send an error. That way we get *some* feedback about it happening to some users AND the user doesn't just get the basic Internal Server Error page.